pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -171,6 +172,22 @@ func (c *Check) Interval() time.Duration {
 	return c.config.MinCollectionInterval
 }
 
+// GetDiagnoses collects diagnoses for diagnose CLI
+func (c *Check) GetDiagnoses() ([]diagnosis.Diagnosis, error) {
```

Comment:
```suggestion
func (c *Check) GetDiagnoses() []diagnosis.Diagnosis {
```

Do we need to return an `error` ? It seems the code never generates an error.

Reply:
Never mind, seems we need to comply with the interface.